input_definition.go

```diff
@@ -254,6 +265,56 @@ type InputDefinitionInfo struct {
 	Fields []InputDefinitionField `json:"fields"`
 }
 
+// Validate the InputDefinitionInfo data
+func (i *InputDefinitionInfo) Validate(columnLabel string) error {
+	numPrimaryKey := 0
+	accountRowID := make(map[string]uint64)
+
+	if len(i.Frames) == 0 {
+		return fmt.Errorf("At least one frame is required per Input Definition")
+	}
+
+	for _, frame := range i.Frames {
+		if err := frame.Validate(); err != nil {
```

Comment:
note that if you do keep the `InputFrame.Validate()` method, it can be simplified to:
```
// Validate the InputFrame data
func (i *InputFrame) Validate() error {
    return ValidateName(i.Name)
}
```
With that said, you may be considering adding a call to the frame option validation in this method as well, which would cause this method to change anyway.